commit 8b0d59fb74b5
Author: Valentin Gosu <valentin.gosu@gmail.com>
Date:   Tue Dec 26 21:47:22 2017 +0100

    Bug 1426996 - Use fallible assign in NS_EscapeAndFilterURL r=kershaw a=gchang
    
    MozReview-Commit-ID: EKVeQb3cboU
    
    --HG--
    extra : source : 8af6e9af76052696204adb50eba2b9bdfe81c149
---
 xpcom/io/nsEscape.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git xpcom/io/nsEscape.cpp xpcom/io/nsEscape.cpp
index 7d072399d3cc..d0ce3f649fce 100644
--- xpcom/io/nsEscape.cpp
+++ xpcom/io/nsEscape.cpp
@@ -446,7 +446,9 @@ NS_EscapeAndFilterURL(const nsACString& aStr, uint32_t aFlags,
   }
 
   if (!appended) {
-    aResult = aStr;
+    if (!aResult.Assign(aStr, fallible)) {
+      return NS_ERROR_OUT_OF_MEMORY;
+    }
   }
 
   return rv;
